esp/src/src-react/components/forms/RenameFile.tsx

```diff
@@ -38,42 +42,71 @@ export const RenameFile: React.FunctionComponent<RenameFileProps> = ({
 
     const { handleSubmit, control, reset } = useForm<RenameFileFormValues>({ defaultValues });
 
+    const service = useConst(() => new FileSprayService({ baseUrl: "" }));
+
     const closeForm = React.useCallback(() => {
         setShowForm(false);
     }, [setShowForm]);
 
     const onSubmit = React.useCallback(() => {
         handleSubmit(
-            (data, evt) => {
-                if (logicalFiles?.length > 0) {
-                    if (logicalFiles?.length === 1) {
-                        const request = { ...data, srcname: logicalFiles[0] };
-                        FileSpray.Rename({ request: request }).then(response => {
-                            closeForm();
-                            if (window.location.hash.match(/#\/files\//) === null) {
-                                if (refreshGrid) refreshGrid(true);
-                            } else {
-                                replaceUrl(`/files/${data.dstname}`);
-                            }
-                        });
-                    } else {
-                        logicalFiles.forEach((logicalFile, idx) => {
-                            const request = { ...data, srcname: logicalFile, dstname: data.targetRenameFile[idx].name };
-                            const requests = [];
-                            requests.push(FileSpray.Rename({ request: request }));
-                            Promise.all(requests).then(_ => {
-                                closeForm();
-                                if (refreshGrid) refreshGrid(true);
-                            });
-                        });
+            async (data, evt) => {
+                const renameRequests = [];
+                const getDfuWuRequests = [];
+
+                if (logicalFiles?.length === 1) {
+                    const request = { ...data, srcname: logicalFiles[0], DFUServerQueue: "" };
```

Comment:
Is `data.targetRenameFile` an array or a string here?

Reply:
It was undefined. I don't recall why, but if there was one file the input was named dstname, and if there was a list, it used the targetRenameFile array. It only uses the targetRenameFile array now.